open-grant-proposals/open-proposal-Improve-stability-of-miner.md

```diff
@@ -0,0 +1,114 @@
+# Open Grant Proposal: Improve the stability of the lotus miner
+
+**Name of Project:** Improve the stability of lotus miner
+
+**Proposal Category:** `core-dev`
+
+**Proposer:** `sirius`
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** "Yes"
+
+# Project Description
+
+The project, which will be developed based on the current lotus source code, is intend to improve the stability of miner in WindowPoSt and mining blocks.
+
+The idea is to take lotus-worker rather then lotus-miner as the entrance  while storing cached or sealed sector files. on the mean time, the lotus-worker should provide services which includes VanillaPoSt of Window and Winning. so that the storage capacity can be expanded via lotus-worker directly.
+
+The logic of PoSt in filecoin-ffi, which mainly includes the distribution of partition calculations, worker scheduling and the collection of the results, is moved to sector-storage
+
+Lotus-worker should supports WindowPoSt and WinningPoSt, which are split from lotus-miner
+
+The details of the solution are as follows: 
+
+- Add WindowPoSt and WinningPoSt seal task types.
+- The storage mounted by lotus-worker can support CanStore through parameter configuration.
+- Extend lotus-worker to support WindowPoSt and WinningPoSt task types through parameter configuration.
+- Extend filecoin-ffi and rust-fil-proof to support getting VanillaPoSt result of Window and Winning and calculating WindowPoSt and WinningPoSt based on VanillaPoSt result.
+- Add calculation WindowPoSt and WinningPoSt interface for WorkerCalls.
+- Extend index to support global acquisition of SectorStorageInfo and getting sector's url.
+- Optimize pubSectorToPriv function to support global acquisition of sector path.
+- Extend http_handler of remote to support Vanilla PoSt of Window and Winning.
+- Considering real-time performance, independent scheduling of WindowPoSt and WinningPoSt.
```

Comment:
We did some thinking about how to handle PoSt scheduling / workers some time ago, and the general idea was that we'd create an entirely new worker type which would only do PoSt, and it wouldn't use the sealing scheduler at all (just use storage for reading challenges, then have it's own much simpler scheduler for PoSt - it can be much simpler because we don't need to worry about storage access, and we don't really need to do complicated resource accounting)

Reply:
Yes, the idea of PoSt scheduling is the same, and its scheduling mainly focuses on GPU resources